yarn-project/archiver/src/archiver/archiver.test.ts

```diff
@@ -129,6 +137,30 @@ function makeContractDeployedEvent(l1BlockNum: bigint, l2BlockNum: bigint) {
   } as unknown as Log<bigint, number, undefined, typeof UnverifiedDataEmitterAbi, 'ContractDeployed'>;
 }
 
+/**
+ * Makes a fake L1ToL2 MessageAdded event for testing purposes.
+ * @param l1BlockNum - L1 block number.
+ * @param l2BlockNum - L2Block number.
+ * @returns An L2BlockProcessed event log.
+ */
+function makeL1ToL2MessageAddedEvent(l1BlockNum: bigint, l2BlockNum: bigint) {
+  return {
+    blockNumber: l1BlockNum,
+    args: {
+      sender: EthAddress.random().toString(),
+      senderChainId: 1n,
+      recipient: AztecAddress.random().toString(),
+      recipientVersion: 1n,
+      content: '0x' + randomBytes(32).toString('hex'),
+      secretHash: '0x' + randomBytes(32).toString('hex'),
+      deadline: 100,
+      fee: 1n,
+      entryKey: '0x' + randomBytes(32).toString('hex'),
+    },
+    transactionHash: `0x${l2BlockNum}`,
+  } as unknown as Log<bigint, number, undefined, typeof RollupAbi, 'L2BlockProcessed'>;
```

Comment:
surprised `L2BlockProcessed` the casted log type? Should it be `MessageAdded` ?

Reply:
Surprised to see the horrible type casts needed, but they seem to be carried over from before, so doesn't make sense to try and change them in this PR.

Reply:
Well, the `unknown` at least can be removed: https://github.com/AztecProtocol/aztec-packages/pull/648

Reply:
Viem really seems to be really nasty on the casting front